imagersite/imager_images/views.py

```diff
@@ -1,71 +1,139 @@
-from django.shortcuts import render, redirect, get_object_or_404
+from django.shortcuts import redirect, get_object_or_404
 from .models import Album, Photo
 from django.http import Http404
+from django.views.generic.detail import DetailView
+from django.views.generic.list import ListView
+from django.views.generic import CreateView
+from .forms import AlbumForm, PhotoForm
 
 
-def photo_view(request, photo_id):
-    photo = get_object_or_404(Photo, id=photo_id)
+class PhotoView(DetailView):
+    template_name = 'imager_images/photo.html'
+    model = Photo
+    pk_url_kwarg = 'photo_id'
 
-    username = request.user.get_username()
+    def get(self, *args, **kwargs):
+        try:
+            self.photo = get_object_or_404(Photo, id=kwargs['photo_id'])
+        except KeyError:
+            raise Http404
```

Comment:
It is not necessary for you to do this. A `DetailView` will look up a single object based on the identifier you specify. Since you have specified `pk_url_kwarg` and the `Photo` model, the `.get()` method will look up the `Photo` that has the id that matches the `photo_id`.